state_machine/README.md

```diff
@@ -0,0 +1,112 @@
+# kth-rpl-hackathon-2022 State Machine
+Repository hosting the code for the 2022 Robotics, Perception and Learning hackathon
+
+## installation
+### Step 1: Install ROS
+For ubuntu 20.04, it is recommended to install ROS Noetic. follow the website [http://wiki.ros.org/noetic/Installation](http://wiki.ros.org/noetic/Installation)
+
+1. Configure your Ubuntu repositories
+Configure your Ubuntu repositories to allow "restricted," "universe," and "multiverse." You can follow the Ubuntu guide for instructions on doing this.
+
+2. Setup your sources.list
+Setup your computer to accept software from packages.ros.org. 
+    ```
+    sudo sh -c 'echo "deb http://packages.ros.org/ros/ubuntu $(lsb_release -sc) main" > /etc/apt/sources.list.d/ros-latest.list'
+    ```
+
+
+3. Set up your keys
+    ```
+    sudo apt install curl # if you haven't already installed curl
+    curl -s https://raw.githubusercontent.com/ros/rosdistro/master/ros.asc | sudo apt-key add -
+    ```
+
+4. 
+    ```
+    sudo apt update
+    ```
+
+5. Now pick how much of ROS you would like to install.
+
+    Desktop-Full Install: (Recommended) : Everything in Desktop plus 2D/3D simulators and 2D/3D perception packages
+
+    ```
+    sudo apt install ros-noetic-desktop-full
+    ```
+
+6. 
+    ```
+    source /opt/ros/noetic/setup.bash
+    ```
+7. Dependencies for building packages
+    ```
+    $ sudo apt install python3-rosdep python3-rosinstall python3-rosinstall-generator python3-wstool build-essential
```

Comment:
as far as I saw, most of your python files are executed by python 2...

Reply:
uhmmm to publish a topic as just a singleton its fine to be python3